examples/next/pages/ui/hooks/useDataStore/filter/CollectionAndRecord.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { User, UserPreference } from '../models';
+import {
+  createDataStorePredicate,
+  getOverrideProps,
+  useDataStoreBinding,
+} from '@aws-amplify/ui-react/internal';
+import { Button, Collection, Flex } from '@aws-amplify/ui-react';
+import { colorFilterPredicate, userFilterPredicate } from '../utils';
+
+const CollectionAndRecord = (props) => {
+  const {
+    backgroundColor,
+    buttonColor: buttonColorProp,
+    items,
+    overrideItems,
+    overrides,
+    ...rest
+  } = props;
+
+  const userFilter = createDataStorePredicate(userFilterPredicate);
```

Comment:
Probably not super impactful for this particular test, since this component doesn't have a parent that could trigger multiple re-renders, but moved either way, it's good practice, thanks for calling it out!